lib/spack/spack/repo.py

```diff
@@ -1397,12 +1397,15 @@ def get_pkg_class(self, pkg_name: str) -> Type["spack.package_base.PackageBase"]
 
         class_name = nm.pkg_name_to_class_name(pkg_name)
 
+        if not self.exists(pkg_name):
+            raise UnknownPackageError(fullname, self)
+
         try:
             module = importlib.import_module(fullname)
         except ImportError as e:
```

Comment:
I don't get why this is special cased. The exceptions that are raised are just any type of exception you get when executing the module. ImportError can be raised because the module itself cannot be imported, but also because during module execution another `import` statement fails. So, I think you should just catch `Exception`.

Reply:
I tried that but it resulted in a number of failures. There are places in the code where it not finding a package means the processing will skip the (in install_test, solver/asp, solver/input_analysis, repo).

Do you mean catch the exception and raise `RepoError`? (I'm currently collecting a list of the unit tests that will fail as a result.)

Reply:
🤔 Perhaps I misunderstood your point. Are you saying to keep lines 1400-1402 but drop catching the `ImportError` exception? 

(I'm trying that now.)